src/block/styled-components.js

```diff
@@ -8,297 +8,243 @@ LICENSE file in the root directory of this source tree.
-    }
-
-    if ($minHeight !== undefined) {
-      styles.minHeight = $minHeight;
-    }
-
-    if ($maxHeight !== undefined) {
-      styles.maxHeight = $maxHeight;
-    }
-
-    if ($overflow === 'scrollX') {
-      styles.overflowX = 'scroll';
-    } else if ($overflow === 'scrollY') {
-      styles.overflowY = 'scroll';
-    } else if ($overflow !== undefined) {
-      styles.overflow = $overflow;
-    }
-
-    if ($margin !== undefined) {
-      styles.margin = sizing[$margin] || $margin;
-    }
-
-    if ($marginTop !== undefined) {
-      styles.marginTop = sizing[$marginTop] || $marginTop;
-    }
-
-    if ($marginRight !== undefined) {
-      styles.marginRight = sizing[$marginRight] || $marginRight;
-    }
-
-    if ($marginBottom !== undefined) {
-      styles.marginBottom = sizing[$marginBottom] || $marginBottom;
-    }
-
-    if ($marginLeft !== undefined) {
-      styles.marginLeft = sizing[$marginLeft] || $marginLeft;
-    }
-
-    if ($padding !== undefined) {
-      styles.padding = sizing[$padding] || $padding;
-    }
-
-    if ($paddingTop !== undefined) {
-      styles.paddingTop = sizing[$paddingTop] || $paddingTop;
-    }
-
-    if ($paddingRight !== undefined) {
-      styles.paddingRight = sizing[$paddingRight] || $paddingRight;
-    }
-
-    if ($paddingBottom !== undefined) {
-      styles.paddingBottom = sizing[$paddingBottom] || $paddingBottom;
-    }
-
-    if ($paddingLeft !== undefined) {
-      styles.paddingLeft = sizing[$paddingLeft] || $paddingLeft;
-    }
-
-    if ($placeContent !== undefined) {
-      styles.placeContent = $placeContent;
-    }
-
-    if ($placeItems !== undefined) {
-      styles.placeItems = $placeItems;
-    }
-
-    if ($placeSelf !== undefined) {
-      styles.placeSelf = $placeSelf;
-    }
-
-    if ($flexWrap !== undefined) {
-      styles.flexWrap = 'wrap';
-    }
-
-    if ($left !== undefined) {
-      styles.left = $left;
-    }
-
-    if ($top !== undefined) {
-      styles.top = $top;
-    }
-
-    if ($right !== undefined) {
-      styles.right = $right;
-    }
-
-    if ($bottom !== undefined) {
-      styles.bottom = $bottom;
-    }
-
-    return styles;
-  },
-);
+type ApplyParams = {
+  property: string,
+  value?: * | Array<*>,
+  // eslint-disable-next-line flowtype/no-weak-types
+  transform?: Function,
+};
+
+function build(breakpoints: BreakpointsT) {
+  const styles = {};
+  const mediaQueries = Object.keys(breakpoints).map(
+    size => `@media screen and (min-width: ${breakpoints[size]}px)`,
+  );
+
+  return {
+    apply: ({property, transform = x => x, value}: ApplyParams) => {
+      if (value == null) {
+        return;
+      }
+
+      if (Array.isArray(value)) {
+        value.forEach((v, index) => {
+          // Do not create a media query for the smallest breakpoint.
+          if (index === 0) {
+            styles[property] = transform(v);
+            return;
+          }
+
+          const mediaQuery = mediaQueries[index];
```

Comment:
It seems like we never create a media query for the `small` breakpoint that makes its value unused and leaves only 2 possible default breakpoints. 
How about we consider the first value in the array as default (the smallest, as we currently do) but we don't skip the first media query in `mediaQueries`? So we'll get 3 media queries in addition to the default (considered smallest screen) value. Something like this:
```
height: '100px', 
[smallBreakpoint]: {height: '200px', }
[mediumBreakpoint]: {height: '300px', }
[largeBreakpoint]: {height: '400px', }
```

Unless I missed something while reading the code 🤔

Reply:
yes - you're totally right. my mistake